Core/StateManager.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace Core
+{
+    public class StateManager
+    {
+        public static SortedSet<Transaction> LoadedState { get; set; } = new SortedSet<Transaction>();
```

Comment:
Don't duplicate state here, it's already present in State class. Call state manager in State.Init to load old state.